core/src/main/java/com/epam/esm/jpa/specification/CertificateSpecification.java

```diff
@@ -0,0 +1,88 @@
+package com.epam.esm.jpa.specification;
+
+import com.epam.esm.model.dto.search.GiftSearchDto;
+import com.epam.esm.model.entity.GiftCertificateEntity;
+import com.epam.esm.util.DaoConstants;
+import com.epam.esm.util.SearchConstants;
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+
+public class CertificateSpecification {
+    public static Specification<GiftCertificateEntity> bySearchRequest(GiftSearchDto giftSearchDto) {
+        String giftNamePrefix = giftSearchDto.getNamePrefix();
+        String giftDescriptionPrefix = giftSearchDto.getDescriptionPrefix();
+        List<String> tagNamePrefixes = giftSearchDto.getTagNamePrefixes();
+
+        Specification<GiftCertificateEntity> finalSpec = (Specification<GiftCertificateEntity>) (root, query, criteriaBuilder) -> null;
+
+        if (giftNamePrefix != null) {
```

Comment:
This specification doesn't work in the same way as it works before
You're re-initializing specification every time if giftNamePrefix, giftDescriptionPrefix or tagNamePrefixes are not null
As a result, you can search certification only by last not null field!